plottr/apps/inspectr.py

```diff
@@ -169,6 +169,35 @@ def setRuns(self, selection: Dict[int, Dict[str, str]]) -> None:
         for i in range(len(self.cols)):
             self.resizeColumnToContents(i)
 
+    def updateRuns(self, selection: Dict[int, Dict[str, str]]) -> None:
+
+        run_added = False
```

Comment:
I think there is scope for simplifying this somewhat. There is a method called findItems (used elsewhere in this file)

Using that with `self.findItems(str(runId),  QtCore.Qt.MatchExactly)` you will get a list or rows with that runId. 

Now if that list is empty you call `self.addRun` 
If the list has one element you check the text in that element and modify it if needed as you currently do
If there are more than one element something is wrong and we should raise an error